entrypoint.sh

```diff
@@ -51,27 +51,51 @@ case ${RUNNER_SCOPE} in
     ;;
 esac
 
-if [[ -n "${ACCESS_TOKEN}" ]]; then
-  _TOKEN=$(bash /token.sh)
-  RUNNER_TOKEN=$(echo "${_TOKEN}" | jq -r .token)
+# If the variable is not set, set it with the default value
+if [ -z "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}" ]; then
+  CONFIGURED_ACTIONS_RUNNER_FILES_DIR="/actions-runner-files"
 fi
 
-echo "Configuring"
-./config.sh \
-    --url "${_SHORT_URL}" \
-    --token "${RUNNER_TOKEN}" \
-    --name "${_RUNNER_NAME}" \
-    --work "${_RUNNER_WORKDIR}" \
-    --labels "${_LABELS}" \
-    --runnergroup "${_RUNNER_GROUP}" \
-    --unattended \
-    --replace
+# Loading the files from the mounted directory
+if [ -d "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}" ]; then
+  cp -p -r "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}/." "/actions-runner"
+fi
 
-unset RUNNER_TOKEN
+if [ -f "/actions-runner/.runner" ]; then
```

Comment:
This will mean that it always retains state and its no longer optional.

Others aside from myself want this to be an opt in functionality

Reply:
I think it makes sense to add 2 new vars total:

`CONFIGURED_ACTIONS_RUNNER_FILES_DIR` as you have, as well as `RUNNER_RETAIN_ENABLE` which is the behavior flag as true/false

Reply:
As I answered:

> "Opt-out" is a quite simple: do not mount a directory and a state will stay inside.

Could you please provide a use case when you mount the directory with the state but disables any logic of communicating with this mounted directory and the state.

If there is no such a use case then according to "Occam's razor" only one var is enough

Reply:
Moreover, I do not know what happens if try to configure when `/actions-runner/.runner` is presented. I do not think it is a good idea.

Reply:
Implicit vs explicit for opt-in

The presense of a directory for a behavior change is implicit vs explicitly saying they want the behavior

Reply:
opt-in:
```
ABC...
volumes:
      ## For avoiding reregistering
      - '${LOCAL_DIRECTORY_FOR_ACTIONS_RUNNER_FILES}:${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}'
      - DEF...
```

opt-out:
```
ABC...
volumes:
      - DEF...
```
or
```
ABC...
volumes:
      ## For avoiding reregistering
      #- '${LOCAL_DIRECTORY_FOR_ACTIONS_RUNNER_FILES}:${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}'
      - DEF...
```

invalid (using null):
```
ABC...
volumes:
      ## For avoiding reregistering
      - '${LOCAL_DIRECTORY_FOR_ACTIONS_RUNNER_FILES}:
      - DEF...
```

As for me, I think that mounting is quite implicit: You mount the state directory - you use the state management; you do not mount the state directory - you do not use state management.

In turn, a user when two variables:
> - I enabled `RUNNER_RETAIN_ENABLE` from supported envvar list
> - The system says `cp:  cannot stat '/actions-runner-files': No such file or directory`
> - ???

or

> - I enabled `RUNNER_RETAIN_ENABLE` from supported envvar list
> - The system says `You do not mount the directory with the state`
> - What is the directory ?

or

> - I mounted the directory with state
> - (The container worked some time) Strange... The state was not saved, why?
> - (Best case) Oh, it appears that you must/I forgot enable `RUNNER_RETAIN_ENABLE`.
> - (Worst case) This thing does not work. I do not have to research or asks for help so I will try another solution